core/templates/pages/preferences-page/preferences-page.component.spec.ts

```diff
@@ -98,4 +232,32 @@ describe('Preferences Controller', function() {
         ['id', 'text']);
     }
   });
+
+  it('should edit profile picture modal', function() {
+    var newPicture = 'new-picture.png';
+    spyOn(mockWindow.location, 'reload').and.callThrough();
+    spyOn($uibModal, 'open').and.returnValue({
+      result: $q.resolve(newPicture)
+    });
+    spyOn(UserService, 'setProfileImageDataUrlAsync').and.returnValue(
+      $q.resolve());
+
+    ctrl.showEditProfilePictureModal();
+    $rootScope.$apply();
+    $rootScope.$apply();
+
+    expect(mockWindow.location.reload).toHaveBeenCalled();
```

Comment:
Can we also make sure that the profile picture is actually set?

Reply:
I don't think so. The profile picture will be only set after the window is reloaded. It's not set before.

Reply:
Hmmm, ok. I think that is being captured in the e2e tests for preferences, so this looks ok.